lib/run.js

```diff
@@ -4,18 +4,44 @@ const EE = require('events').EventEmitter
 const URL = require('url')
 const hdr = require('hdr-histogram-js')
 const timestring = require('timestring')
+const reInterval = require('reinterval')
 const Client = require('./httpClient')
 const DefaultOptions = require('./defaultOptions')
 const multipart = require('./multipart')
-const histUtil = require('hdr-histogram-percentiles-obj')
-const reInterval = require('reinterval')
+const { isMainThread } = require('./worker_threads')
 const { ofURL, checkURL } = require('./url')
 const { parseHAR } = require('./parseHAR')
-const histAsObj = histUtil.histAsObj
-const addPercentiles = histUtil.addPercentiles
+const initWorkers = require('./manager')
+const aggregateResult = require('./aggregateResult')
+const printResult = require('./printResult')
+
+function handleFinish (result, opts, cb) {
+  if (cb) cb(result)
+  if (opts.json) {
+    console.log(JSON.stringify(result))
+  }
+
+  // the code below this `if` just renders the results table...
+  // if the user doesn't want to render the table, we can just return early
+  if (!isMainThread || opts.renderResultsTable === false) return
+
+  printResult(result, opts)
+}
 
 function run (opts, cb) {
-  return _run(opts, cb)
+  if (opts.useWorkers && isMainThread) {
```

Comment:
Can we avoid using `isMainThread` and pass around options instead?

Reply:
Sorry not sure I understand what you mean. using `isMainThread` here ensures that workers do not try and initialize more workers. How would you do it?

Reply:
How is it possible for a worker to enter this function?

Reply:
worker calls this from [here](https://github.com/mcollina/autocannon/pull/290/files#diff-df9477f338c214ed04a5e566edd666461eee57595f4b8800d286075824c2ad2cR7).